### ISSUE TITLE
Cannot set Object from Variant ByRef

### ISSUE THREAD

**Issue by cristianbuse** [WaynePhillipsEA/twinbasic#202]
**Describe the bug**
Runtime error on redirected Object stored in array.

**To Reproduce**
```VBA
Module Test
    Public Sub Main()
        Sub1 Array(Err)
        
        Stop
    End Sub

    Public Sub Sub1(ByRef arr As Variant)
        Sub2 arr(0)
    End Sub

    Public Sub Sub2(ByRef v As Variant)
        Dim obj As Object

        Set obj = v                     'Runtime error
    End Sub
End Module
```

**Desktop (please complete the following information):**
 - OS: Windows 10 Enterprise V1909
 - twinBASIC compiler v0.10.1204

**Additional Context**
Seems related to #182

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.1386, thanks for reporting.

**Comment by cristianbuse** [comment]
If we replace ```Dim obj As Object``` with ```Dim obj As IUnknown``` then same issue occurs

**Comment by WaynePhillipsEA** [comment]
Apologies... missed that one.

**Comment by cristianbuse** [comment]
No need to apologize. One can't think about every single edge case. Thanks for always fixing them so promptly!

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.10.1389.  Thanks!
